source/test-suite/Main.hs

```diff
@@ -1070,178 +1071,183 @@ main = Hspec.hspec . Hspec.parallel . Hspec.describe "cabal-gild" $ do
+      "library\n -- cabal-gild: discover\n exposed-modules:"
+      "library\n  -- cabal-gild: discover\n  exposed-modules: M\n"
 
   Hspec.it "discovers a .x file" $ do
     expectDiscover
-      [(".", ["M.x"])]
-      "library\n -- cabal-gild: discover .\n exposed-modules:"
-      "library\n  -- cabal-gild: discover .\n  exposed-modules: M\n"
+      [["M.x"]]
+      "library\n -- cabal-gild: discover\n exposed-modules:"
+      "library\n  -- cabal-gild: discover\n  exposed-modules: M\n"
 
   Hspec.it "discovers a .cpphs file" $ do
     expectDiscover
-      [(".", ["M.cpphs"])]
-      "library\n -- cabal-gild: discover .\n exposed-modules:"
-      "library\n  -- cabal-gild: discover .\n  exposed-modules: M\n"
+      [["M.cpphs"]]
+      "library\n -- cabal-gild: discover\n exposed-modules:"
+      "library\n  -- cabal-gild: discover\n  exposed-modules: M\n"
 
   Hspec.it "discovers a .hsig file" $ do
     expectDiscover
-      [(".", ["M.hsig"])]
-      "library\n -- cabal-gild: discover .\n exposed-modules:"
-      "library\n  -- cabal-gild: discover .\n  exposed-modules: M\n"
+      [["M.hsig"]]
+      "library\n -- cabal-gild: discover\n exposed-modules:"
+      "library\n  -- cabal-gild: discover\n  exposed-modules: M\n"
 
   Hspec.it "discovers a .lhsig file" $ do
     expectDiscover
-      [(".", ["M.lhsig"])]
-      "library\n -- cabal-gild: discover .\n exposed-modules:"
-      "library\n  -- cabal-gild: discover .\n  exposed-modules: M\n"
+      [["M.lhsig"]]
+      "library\n -- cabal-gild: discover\n exposed-modules:"
+      "library\n  -- cabal-gild: discover\n  exposed-modules: M\n"
 
   Hspec.it "discovers a signature" $ do
     expectDiscover
-      [(".", ["S.hsig"])]
-      "library\n -- cabal-gild: discover .\n signatures:"
-      "library\n  -- cabal-gild: discover .\n  signatures: S\n"
+      [["S.hsig"]]
+      "library\n -- cabal-gild: discover\n signatures:"
+      "library\n  -- cabal-gild: discover\n  signatures: S\n"
 
   Hspec.it "ignores discover pragma separated by comment" $ do
     expectGilded
-      "library\n -- cabal-gild: discover .\n -- foo\n exposed-modules: M"
-      "library\n  -- cabal-gild: discover .\n  -- foo\n  exposed-modules: M\n"
+      "library\n -- cabal-gild: discover\n -- foo\n exposed-modules: M"
+      "library\n  -- cabal-gild: discover\n  -- foo\n  exposed-modules: M\n"
 
   Hspec.it "ignores misplaced discover pragma" $ do
     expectGilded
-      "-- cabal-gild: discover .\nname: p"
-      "-- cabal-gild: discover .\nname: p\n"
+      "-- cabal-gild: discover\nname: p"
+      "-- cabal-gild: discover\nname: p\n"
 
   Hspec.it "ignores unknown pragma" $ do
     expectGilded
       "-- cabal-gild: unknown"
       "-- cabal-gild: unknown\n"
 
+  Hspec.it "discovers from the currently directory explicitly" $ do
+    expectDiscover
+      [["M.hs"]]
+      "library\n -- cabal-gild: discover .\n exposed-modules:"
+      "library\n  -- cabal-gild: discover .\n  exposed-modules: M\n"
```

Comment:
This is a new test since I changed all the existing `discover .` to just `discover`.